src/Foundatio.Repositories.Elasticsearch/Configuration/VersionedIndex.cs

```diff
@@ -21,6 +22,16 @@ public VersionedIndex(IElasticConfiguration configuration, string name, int vers
         public int Version { get; }
         public string VersionedName { get; }
         public bool DiscardIndexesOnReindex { get; set; } = true;
+        private List<ReindexScript> ReindexScripts { get; } = new List<ReindexScript>();
+    
+        private class ReindexScript {
+            public Func<int, int, bool> ShouldRun { get; set; }
```

Comment:
This is a bit confusing as it is. Can we at least create a delegate for this so its more intuitive as to what the parameters are?